### ISSUE TITLE
Allow map to fail

### ISSUE THREAD

**Issue by InnovativeInventor** [dry-python/returns#403]
Is there currently any way to map a Maybe object to a function to execute on fail?

Current code:
```python
def benchmark(strategies: Maybe[list] = None) -> Maybe[axelrod.result_set.ResultSet]:
#     Maybe.from_value(strategies).map(compare)
    Maybe.new(strategies).map(compare)
```

Suggested:
```python
def benchmark(strategies: Maybe[list] = None) -> Maybe[axelrod.result_set.ResultSet]:
#     Maybe.from_value(strategies).map(compare, fail)
    Maybe.new(strategies).map(compare, fail)
```

Awesome project though – FP in Python!

**Comment by thepabloaguilar** [comment]
Currently, if you want to execute something in case of `Maybe` container is `Nothing`, you can check:

```python
def benchmark(strategies: Maybe[list] = None) -> Maybe[axelrod.result_set.ResultSet]:
    maybe_strategies = Maybe.new(strategies).map(compare)
    if maybe_strategies == Nothing:
        fail(...)
```

`Maybe` container doesn't have any method to apply a function when it's `Nothing` because if the container is `Nothing` there isn't any value and don't make any sense you apply a function to nothing. If you need, you have to explicitly check if it's `Nothing`.

**Comment by sobolevn** [comment]
`master` version has `or_else_call` do achieve what you want. It is going to be released as `0.14` pretty soon!

**Comment by sobolevn** [comment]
Closing for now, please feel free to ask any questions you have! 👍
